### PR TITLE
Add auto-remove-stale-data to json schema - (hot-restart) persistence

### PR DESCRIPTION
**Pull request by fbarotov**
Fixes https://github.com/hazelcast/hazelcast/issues/19373

Checklist:
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

File: hazelcast/src/main/resources/hazelcast-config-5.0.json [hunk 1]
```diff
@@ -3281,6 +3281,10 @@
           "type": "boolean",
           "default": false
         },
+        "auto-remove-stale-data": {
+          "type": "boolean",
+          "default": false
```
Comment (vbekiaris) [inline hazelcast/src/main/resources/hazelcast-config-5.0.json:3286]: The default value is `true`. Also, you could copy the [documentation from the XSD](https://github.com/hazelcast/hazelcast/blob/462045ca037af733bded32375ea4d7e96b52628a/hazelcast/src/main/resources/hazelcast-config-5.1.xsd#L3927-L3935) into the json schema `"description"` field, but now I see that the documentation has to be updated to reflect changes we made in 5.0 persistence, so feel free to create a separate issue for that.

**Review by vbekiaris — CHANGES_REQUESTED** [review]
Something to consider: is there an automated way to avoid skipping config elements from the json schema in the future?

File: hazelcast/src/main/resources/hazelcast-config-5.0.json (hunk 1 above)
Comment (fbarotov) [reply hazelcast/src/main/resources/hazelcast-config-5.0.json]: Thanks for the catch and comment :) I [opened](https://github.com/hazelcast/hazelcast/issues/19708) an issue regarding (config) documentation update on persistence

**Comment by fbarotov** [comment]
> Something to consider: is there an automated way to avoid skipping config elements from the json schema in the future?

I thought about it, and the only way seems to have an XSD (to and from) Json-schema converter, where we keep one to one mapping. However, I could not find such a Java library that support both way conversion and go beyond basic level support

**Review by vbekiaris — APPROVED** [review]

**Pull request merged**